src/main/kotlin/com/chattriggers/ctjs/api/world/TabList.kt

```diff
@@ -102,17 +123,277 @@ object TabList {
      */
     @JvmStatic
     fun setFooter(footer: Any?) {
+        customFooter = false
         when (footer) {
-            is CharSequence -> toMC()?.setFooter(TextComponent(footer))
-            is TextComponent -> toMC()?.setFooter(footer)
-            is Text -> toMC()?.setFooter(footer)
-            null -> toMC()?.setFooter(null)
+            is TextComponent -> {
+                tabListHeader = footer
+                toMC()?.setFooter(footer)
+            }
+            is CharSequence, is Text -> {
+                tabListHeader = TextComponent(footer)
+                toMC()?.setFooter(tabListHeader)
+            }
+            null -> {
+                tabListHeader = TextComponent("")
+                toMC()?.setFooter(null)
+            }
         }
+        customFooter = true
     }
 
     @JvmStatic
     fun clearFooter() = setFooter(null)
 
+    /**
+     * Gets names set in scoreboard objectives
+     *
+     * @return The formatted names
+     */
+    @JvmStatic
+    fun getNamesByObjectives(): List<String> {
+        val scoreboard = Scoreboard.toMC() ?: return emptyList()
+        val tabListObjective = getList() ?: return emptyList()
+
+        val scores = scoreboard.getScoreboardEntries(tabListObjective)
+
+        return scores.map {
+            val team = scoreboard.getTeam(it.owner)
+            TextComponent(MCTeam.decorateName(team, TextComponent(it.owner))).formattedText
+        }
+    }
+
+    /**
+     * Get all names on the tab list
+     *
+     * @return the list of names
+     */
+    @JvmStatic
+    fun getNames(): List<Name> {
+        if (needsUpdate) {
+            updateNames()
+            needsUpdate = false
+        }
+
+        return tabListNames
+    }
+
+    /**
+     * Gets all names in tabs without formatting
+     *
+     * @return the unformatted names
+     */
+    @JvmStatic
+    fun getUnformattedNames(): List<String> {
+        if (needsUpdate) {
+            updateNames()
+            needsUpdate = false
+        }
+
+        return tabListNames.map { it.toMC().profile.name }
+    }
+
+    /**
+     * Adds a new name to the tab list
+     *
+     * @param name the formatted name to add
+     * @param useExistingSkin whether to use the skin of the associated Minecraft account using [name]
```

Comment:
What happens if this flag is false? It's not really clear from the doc or the code tbh.

Reply:
If the flag is false, then it uses one of the dozen or so default skins (Alex, Steve, etc), instead of the one associated with the account with that name